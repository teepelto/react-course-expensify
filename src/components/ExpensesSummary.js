import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedTotal = numeral(expensesTotal).format('$0,0.00');
    
    return (
        <div>
            <h1>
                Viewing {expenseCount} {expenseWord} totalling {formattedTotal}.
            </h1>
        </div>
        
    );
};

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: expenses.length,
        expensesTotal: selectExpensesTotal(expenses) / 100
    };
};

export default connect(mapStateToProps)(ExpensesSummary);