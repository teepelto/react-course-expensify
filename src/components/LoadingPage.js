import React from 'react';

const LoadingPage = () => (
    <div class="loader"></div>    
);

export default LoadingPage;