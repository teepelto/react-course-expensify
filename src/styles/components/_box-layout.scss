.box-layout {
    background-color: $primary-blue;
    background-size: cover;
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;
    
    &__box {
        background-color: fade-out($primary-green, .15);
        border-radius: 3px;
        text-align: center;
        width: 25rem;
        padding: $l-size $m-size;
    }
    
    &__title {
        margin: 0 0 $m-size 0;
        line-height: 1;
    }
}
