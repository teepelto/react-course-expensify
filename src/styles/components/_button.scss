.button {
    color: white;
    background-color: $tertiary-green;
    border: none;
    border-radius: 3px;
    padding: $s-size;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    display: inline-block;
    text-decoration: none;
    
    &--link {
        background: none;
    }
    
    &--secondary {
        background-color: $primary-blue;
    }
}