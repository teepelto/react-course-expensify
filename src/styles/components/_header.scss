.header {
    background: $secondary-blue;
    
    &__content {
        display: flex;
        justify-content: space-between;
        align-items: center;
        padding: $s-size 0;
    }
    
    &__title {
        color: white;
        text-decoration: none;
        margin: 0;
    }
}