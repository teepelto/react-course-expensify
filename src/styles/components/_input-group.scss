.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $m-size;

    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        margin-bottom: $l-size;
    }

    &__item {
        margin-bottom: $s-size;
    }

    &__item:not(:last-child) {
        @media (min-width: $desktop-breakpoint) {
            margin-right: $s-size;
        }
    }
    
}