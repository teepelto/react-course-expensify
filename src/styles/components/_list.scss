.list-header {
    background-color: $primary-green;
    border: 1px solid darken($primary-green, 7%);
    color: $secondary-blue;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}

.list-body {
    margin-bottom: $m-size;

    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $l-size;
    }
}

.list-item {
    border: 1px solid darken($primary-green, 7%);
    border-top: none;
    color: $secondary-blue;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none;
    transition: background .3s ease;

    &:hover {
        background: $primary-green;
    }
    
    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
        padding: $m-size;
    }
    
    &__title {
        margin: 0;
        word-break: break-all;
    }
    
    &__sub-title {
        color: $primary-blue;
        font-size: $font-size-small;
    }
    
    &__data {
        margin: $s-size 0 0 0;
        
        @media (min-width: $desktop-breakpoint) {
            margin: 0;
            padding-left: $s-size;
        }
    }
}

.list-item--message {
    align-items: center;
    justify-content: center;
    color: $primary-blue;
    padding: $m-size;

    &:hover {
        background: none;
    }
}