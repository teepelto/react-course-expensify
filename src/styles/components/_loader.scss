.loader {
    border: 16px solid #f3f3f3; /* Light grey */
    border-top: 16px solid $secondary-blue;
    border-radius: 50%;
    width: 120px;
    height: 120px;
    margin: calc(50vh - 60px) auto;
    animation: spin 2s linear infinite;
}

@keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
}