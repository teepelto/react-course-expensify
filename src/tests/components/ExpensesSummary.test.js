/* global expect, jest */
import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={666} />);
    expect(wrapper).toMatchSnapshot();
});
