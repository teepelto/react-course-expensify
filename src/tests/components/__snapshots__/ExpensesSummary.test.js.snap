// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $0.00
    .
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $0.00
    .
  </h1>
</div>
`;
